csharp/extractor/Semmle.Extraction.CSharp/Entities/TypeMention.cs

```diff
@@ -54,19 +63,27 @@ protected override void Populate(TextWriter trapFile)
             switch (syntax.Kind())
             {
                 case SyntaxKind.ArrayType:
+                case SyntaxKind.PointerType:
                     Emit(trapFile, loc ?? syntax.GetLocation(), parent, type);
-                    Create(cx, GetElementType(syntax), this, GetElementType(type));
+                    Create(cx, GetArrayElementType(syntax), this, GetArrayElementType(type));
                     return;
                 case SyntaxKind.NullableType:
                     var nts = (NullableTypeSyntax)syntax;
                     if (type is NamedType nt)
                     {
-                        Emit(trapFile, loc ?? syntax.GetLocation(), parent, type);
-                        Create(cx, nts.ElementType, this, nt.symbol.IsReferenceType ? nt : nt.TypeArguments[0]);
+                        if (!nt.symbol.IsReferenceType)
+                        {
+                            Emit(trapFile, loc ?? syntax.GetLocation(), parent, type);
+                            Create(cx, nts.ElementType, this, nt.TypeArguments[0]);
+                        }
+                        else
+                        {
+                            Create(cx, nts.ElementType, parent, type);
+                        }
                     }
                     else if (type is ArrayType array)
                     {
-                        Create(cx, nts.ElementType, parent, array);
+                        Create(cx, nts.ElementType, parent, type);
```

Comment:
Why this change?

Reply:
This is only cosmetics. This way this branch and the previous one are exactly the same. I'll change the `if` to `if (type is ArrayType)`